testdata/project-v3-v1beta1/go.mod

```diff
@@ -5,14 +5,14 @@ go 1.17
 require (
 	github.com/onsi/ginkgo v1.16.4
 	github.com/onsi/gomega v1.13.0
-	k8s.io/api v0.22.1
```

Comment:
How did this pass CI? v1beta1 is not available in k8s 1.22, right? :confused:

Reply:
Good catcher 🥇 

The projects scaffolds under testdata are not tested via e2e. We just check the scaffold features with them. They are generated here: https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/testdata/generate.sh

We create and scaffold projects to do the e2e tests which are implemented here: https://github.com/kubernetes-sigs/kubebuilder/tree/master/test/e2e

See here: https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/e2e/v3/generate_test.go w do not create a new API using the flag option v1beta1 and try to apply in the k8s versions < 1.22. 

If you wish to create this test that is totally fine but it shows not mandatory since it is a deprecated feature kept only to not break backwards compatibilities.